src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -286,8 +286,6 @@ ClientRequest::recover_missing_snaps(
   ObjectContextRef head,
   std::set<snapid_t> &snaps)
 {
-  co_await ihref.enter_stage<interruptor>(
-    client_pp(*pg).recover_missing_snaps, *this);
```

Comment:
Is this change intentional or it this still WIP? If so, can you please explain? Same for `recover_missing_lock_obc`

Reply:
Yes, it's intentional. The reason is:

1. the pipeline phase in which the obc loading is done has to be entered "synchronouly", however, the leaving of an `OrderedConcurrentPhase` can't be so, which is the cause of the failure in https://github.com/ceph/ceph/pull/57966#pullrequestreview-2124738667
2. on the other hand, the whole missing object/snaps recovery procedure can and should be a `OrderedConcurrentPhase`, so I made the whole procedure encapsulated in the same concurrent `recover_missing` phase

Reply:
This should probably be part of the commit message.
The stages removed here `recover_missing_lock_obc` and `recover_missing_snaps` has no other users and can be removed entirely, right?

Reply:
> This should probably be part of the commit message.
> The stages removed here `recover_missing_lock_obc` and `recover_missing_snaps` has no other users and can be removed entirely, right?

Yes, will do.